src/scriptChallenge.js

```diff
@@ -0,0 +1,153 @@
+        selectedTags = Array.from(selectedTags);
+        selectedTags = selectedTags.map((tag) => {
+          return tag.textContent;
+        });
+        let data = challenges;
+        console.log(data)
+        data = await filterByTags(selectedTags, data);
+        filterAndUpdateCards(
+          data,
+          createChallengeCard,
+          challengeContainer,
+          filterText,
+          filterByTypes,
+          filterByRating
+        );
+      });
+  });
+
+  const tags = getTags(challenges);
+  const buttons = displayTags(tags, challenges, tagParent);
+  
+  //Add eventlistener for tag buttons
+  (await buttons).forEach((button) => {
+    button.addEventListener("click", async (event) => {
+      const currentButton = event.target;
+      currentButton.classList.toggle("active");
+      let selectedTags = tagParent.querySelectorAll(".active");
+      selectedTags = Array.from(selectedTags);
+      selectedTags = selectedTags.map((tag) => {
+        return tag.textContent;
+      });
+      let data = challenges;
+      data = await filterByTags(selectedTags, data);
+      filterAndUpdateCards(
+        data,
+        createChallengeCard,
+        challengeContainer,
+        filterText,
+        filterByTypes,
+        filterByRating
+      );
+    });
+  });
+
+  // Add eventlistener for text filter
+  textFilter.addEventListener("keyup", async () => {
+    let selectedTags = tagParent.querySelectorAll(".active");
+    selectedTags = Array.from(selectedTags);
+    selectedTags = selectedTags.map((tag) => {
+      return tag.textContent;
+    });
+    let data = challenges;
+    data = await filterByTags(selectedTags, data);
+    filterAndUpdateCards(
+      data,
+      createChallengeCard,
+      challengeContainer,
+      filterText,
+      filterByTypes,
+      filterByRating
+    );
+  });
+
+  //Add eventlistener for type online
+  onlineTypeInput.addEventListener("click", async () => {
+    let selectedTags = tagParent.querySelectorAll(".active");
+    selectedTags = Array.from(selectedTags);
+    selectedTags = selectedTags.map((tag) => {
+      return tag.textContent;
+    });
+    let data = challenges;
+    data = await filterByTags(selectedTags, data);
+    filterAndUpdateCards(
+      data,
+      createChallengeCard,
+      challengeContainer,
+      filterText,
+      filterByTypes,
+      filterByRating
+    );
+  });
+
+  //Add eventlistener for type onsite
+  onsiteTypeInput.addEventListener("click", async () => {
+    let selectedTags = tagParent.querySelectorAll(".active");
+    selectedTags = Array.from(selectedTags);
+    selectedTags = selectedTags.map((tag) => {
+      return tag.textContent;
+    });
+    let data = challenges;
+    data = await filterByTags(selectedTags, data);
+    filterAndUpdateCards(
+      data,
+      createChallengeCard,
+      challengeContainer,
+      filterText,
+      filterByTypes,
+      filterByRating
+    );
+  });
+}
```

Comment:
Behövs 2 eventListeners för online/onsite? De ser ut att göra samma sak?

Reply:
aah precis! Kolla in funktionen "handleFilterInput" i filterType.js! Den koden funkar även ihop med html elementen i main branch :)